Add support for generating and installing shell completions

Piccirello opened the pull request:
This supports generating completions for bash and zsh, and installing completions for bash.

Closes DPLR-585

Commit:
Add support for generating and installing shell completions

rgmvisser reviewed (commented):
Is it correct that you have to run a command to enable the completion? Can we have this on by default, or are there some downsides to that?

Piccirello commented:
> Is it correct that you have to run a command to enable the completion? Can we have this on by default, or are there some downsides to that?

That is correct. I'm tracking automated installation in DPLR-1376.

ryan-blunden commented on pkg/cmd/completion.go:
```diff
@@ -16,35 +16,102 @@ limitations under the License.
+
+		utils.Log("Your shell has been configured for Doppler CLI completions!")
+		utils.Log("")
+		utils.LogWarning("You must have the 'bash-completion' package installed for completions to work.")
```
```suggestion
		utils.Log("The homebrew 'bash-completion' package is required completions for to work. See https://docs.brew.sh/Shell-Completion for more info.")
```

ryan-blunden commented on pkg/cmd/completion.go:
```diff
@@ -16,35 +16,102 @@ limitations under the License.
+			utils.HandleError(err, "Unable to write completion file")
+		}
+
+		utils.Log("Your shell has been configured for Doppler CLI completions!")
```
```suggestion
		utils.Log("Your shell has been configured for Doppler CLI completions! Restart your shell to apply.")
```

rgmvisser reviewed (dismissed):
Looks good from my side

ryan-blunden reviewed (dismissed):
This is awesome! 

Consider updating the log messages to make it as obvious and easy as possible for auto-complete to start working, but not wedded to my suggestions.

Piccirello commented:
Both great suggestions, will update.

Piccirello commented:
@ryan-blunden I addressed your comments with some minor modifications. Can you take another look and lmk your thoughts ([diff](https://github.com/DopplerHQ/cli/compare/b696d3d38628a852872f32d12ee2be0c3be34ed9..a876ecd7c8ae4b4ec4c4844a7e99fbd28f773d8f))?

ryan-blunden commented:
Brilliant! Helpful and to the point.
